include/ck/utility/data_type.hpp

```diff
@@ -7,11 +7,22 @@
 
 namespace ck {
 
-using bhalf_t = ushort;
-using half_t  = _Float16;
-using int4_t  = _BitInt(4);
-using f8_t    = _BitInt(8);
-using bf8_t   = unsigned _BitInt(8);
+using half_t = _Float16;
+using int4_t = _BitInt(4);
+using f8_t   = _BitInt(8);
+using bf8_t  = unsigned _BitInt(8);
+
+struct bhalf_t
+{
+    using type      = bhalf_t;
+    using data_type = ushort;
+
+    ushort data;
+
+    __host__ __device__ bhalf_t() = default;
+
+    __host__ __device__ constexpr bhalf_t(ushort init) : data(init) {}
+};
```

Comment:
there will be a discrepancy on data type convert. In old ck we have everything inside ck::type_convert(). But if we use custom struct as data type it's better we take care of all the conversion inside this struct, or we may have change multiple places due to a single data type.
Also the same for INF/NAN/MIN/MAX definition in `ck::NumericLimits`
This could be our next refactor point for ck tile. cc @zjing14

Reply:
Hi @carlushuang, you are totally right that custom types conversion is tricky. In https://github.com/ROCm/composable_kernel/pull/1167 we refactor f8_t and bf8_t as custom types and refactor components to enable their support including conversion.